lib-injection/host_inject.rb

```diff
@@ -1,249 +1,7 @@
-        lockfile = Bundler::SharedHelpers.default_lockfile
-
-        datadog_gemfile = gemfile.dirname + '.datadog-Gemfile'
-        datadog_lockfile = lockfile.dirname + '.datadog-Gemfile.lock'
-
-        # Copies for trial
-        ::FileUtils.cp gemfile, datadog_gemfile
-        ::FileUtils.cp lockfile, datadog_lockfile
-
-        injection_failure = false
-
-        # This is order dependent
-        [
-          'msgpack',
-          'ffi',
-          'datadog-ruby_core_source',
-          'libdatadog',
-          'libddwaf',
-          'datadog'
-        ].each do |gem|
-          fork do
-            $stdout = File.new('/dev/null', 'w')
-            $stderr = File.new('/dev/null', 'w')
-            Bundler::CLI::Common.select_spec(gem)
-          end
-
-          _, status = Process.wait2
-          if status.success?
-            utils.debug "#{gem} already installed... skipping..."
-            next
-          end
-
-          bundle_add_cmd = "bundle add #{gem} --skip-install --version #{gem_version_mapping[gem]} "
-          bundle_add_cmd << ' --verbose ' if ENV['DD_TRACE_DEBUG'] == 'true'
-          bundle_add_cmd << '--require datadog/single_step_instrument' if gem == 'datadog'
-
-          utils.debug "Injection with `#{bundle_add_cmd}`"
-
-          env = { 'BUNDLE_GEMFILE' => datadog_gemfile.to_s,
-                  'DD_TRACE_SKIP_LIB_INJECTION' => 'true',
-                  'GEM_PATH' => utils.path }
-          add_output, add_status = Open3.capture2e(env, bundle_add_cmd)
-
-          if add_status.success?
-            utils.debug "Successfully injected #{gem} into the application."
-          else
-            injection_failure = true
-            utils.error "Injection failed: Unable to add datadog. Error output: #{add_output}"
-          end
-        end
-
-        if injection_failure
-          ::FileUtils.rm datadog_gemfile
-          ::FileUtils.rm datadog_lockfile
-          telemetry.emit(pid, utils.version, [{ name: 'library_entrypoint.error', tags: ['error_type:injection_failure'] }])
-          exit!(1)
-        else
-          write.puts datadog_gemfile.to_s
-          telemetry.emit(pid, utils.version, [{ name: 'library_entrypoint.complete', tags: ['injection_forced:false'] }])
-        end
-      end
-    end
-
-    write.close
-    gemfile = read.read.to_s.chomp
-
-    _, status = Process.wait2
-    ENV['DD_TRACE_SKIP_LIB_INJECTION'] = 'true'
-
-    if status.success?
-      dd_lib_injection_path = utils.path
-
-      Gem.paths = { 'GEM_PATH' => "#{dd_lib_injection_path}:#{ENV['GEM_PATH']}" }
-      ENV['GEM_PATH'] = Gem.path.join(':')
-      ENV['BUNDLE_GEMFILE'] = gemfile
-      utils.debug "Fork success: Using Gemfile `#{gemfile}`"
-    else
-      utils.debug 'Fork abort'
-    end
-  else
-    utils.debug 'Fork not supported... skipping injection'
-    telemetry.emit(pid, utils.version, [{ name: 'library_entrypoint.abort', tags: ['reason:fork_not_supported'] }])
-  end
+if RUBY_VERSION.start_with?('2.5.')
```

Comment:
Should this be `>=` ?

Reply:
@p-datadog is right!

[Link to red vaccine ](https://github.com/TonyCTHsu/vaccine/actions/runs/11956259855/job/33330597924 )

Reply:
Oh my 🤦. I guess that's what you get when writing code 5min before signing off for the day xD